pkg/rainbow/storage/dbram/dbram.go

```diff
@@ -25,12 +25,23 @@ func NewDB() *DB {
 }
 
 func (db *DB) InsertOptions(options []rainbow.Option) error {
-	for _, o := range options {
-		if _, ok := db.optionsByProvider[o.Provider]; !ok {
-			db.optionsByProvider[o.Provider] = []rainbow.Option{}
+	for _, newOpt := range options {
+		provider := newOpt.Provider
+
+		if _, ok := db.optionsByProvider[provider]; !ok {
+			db.optionsByProvider[provider] = []rainbow.Option{}
+		}
+
+		for i, oldOpt := range db.optionsByProvider[provider] {
+			if oldOpt.Name == newOpt.Name && oldOpt.Expiry == newOpt.Expiry {
+				db.optionsByProvider[provider][i] = newOpt
+				goto done
+			}
 		}
 
-		db.optionsByProvider[o.Provider] = append(db.optionsByProvider[o.Provider], o)
+		db.optionsByProvider[provider] = append(db.optionsByProvider[provider], newOpt)
+
+	done:
```

Comment:
Do not use `goto`.

A simple bool to tell if newOpt has been updated or not (line37) should be enough.

Reply:
To avoid the `goto` I have implemented a sub-function.
The code is nicer ;-)
Is it :ok: for you?

Reply:
I prefer this way, this function can be called `InserOption` (without a s at the end).